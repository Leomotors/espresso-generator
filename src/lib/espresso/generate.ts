export function generateIO(bitsInput: number) {
  const keys = Array.from({ length: 2 ** bitsInput }, (_, i) =>
    i.toString(2).padStart(bitsInput, "0"),
  );

  return Object.fromEntries(keys.map((key) => [key, ""]));
}

export function generateOI(bitsOutput: number) {
  const keys = Array.from({ length: bitsOutput }, (_, i) =>
    String.fromCharCode(97 + i),
  );

  return Object.fromEntries(keys.map((key) => [key, ""]));
}
