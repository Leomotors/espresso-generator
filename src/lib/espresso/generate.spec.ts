import { describe, it, expect } from "vitest";
import { generateIO, generateOI } from "./generate";

describe("generate", () => {
  it("generateIO", () => {
    expect(generateIO(2)).toMatchSnapshot();
    expect(generateIO(3)).toMatchSnapshot();
    expect(generateIO(4)).toMatchSnapshot();
  });

  it("generateOI", () => {
    expect(generateOI(2)).toMatchSnapshot();
    expect(generateOI(3)).toMatchSnapshot();
    expect(generateOI(4)).toMatchSnapshot();
  });
});
