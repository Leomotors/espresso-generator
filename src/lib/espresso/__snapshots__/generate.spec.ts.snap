// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generate > generateIO 1`] = `
{
  "00": "",
  "01": "",
  "10": "",
  "11": "",
}
`;

exports[`generate > generateIO 2`] = `
{
  "000": "",
  "001": "",
  "010": "",
  "011": "",
  "100": "",
  "101": "",
  "110": "",
  "111": "",
}
`;

exports[`generate > generateIO 3`] = `
{
  "0000": "",
  "0001": "",
  "0010": "",
  "0011": "",
  "0100": "",
  "0101": "",
  "0110": "",
  "0111": "",
  "1000": "",
  "1001": "",
  "1010": "",
  "1011": "",
  "1100": "",
  "1101": "",
  "1110": "",
  "1111": "",
}
`;

exports[`generate > generateOI 1`] = `
{
  "a": "",
  "b": "",
}
`;

exports[`generate > generateOI 2`] = `
{
  "a": "",
  "b": "",
  "c": "",
}
`;

exports[`generate > generateOI 3`] = `
{
  "a": "",
  "b": "",
  "c": "",
  "d": "",
}
`;
